packages/terra-functional-testing/src/services/wdio-visual-regression-service/methods/BaseCompare.js

```diff
@@ -1,27 +1,128 @@
 /* eslint-disable class-methods-use-this, no-unused-vars */
+import path from 'path';
+
+
+const TEST_ID_REGEX = /\[([^)]+)\]/;
 
 export default class BaseCompare {
+  constructor(options) {
+    const {
+      baseScreenshotDir,
+      locale,
+      theme,
+    } = options;
+
+    // screenshot naming config
+    this.baseScreenshotDir = baseScreenshotDir || process.cwd();
+    this.locale = locale || 'en';
+    this.theme = theme || 'terra-default-theme';
+  }
+
   /**
    * You can do here your image comparison magic.
+   * @param {Object} context - information provided to process the screenshot
+   * @param {Object} context.browser - { name, version, userAgent }
+   * @param {Object} context.suite - the test suite that is running
+   * @param {Object} context.test - the test that is running
+   * @param {Object} context.meta - { element, exclude, hide, remove, viewport}
+   * @param {*} base64Screenshot - the screenshot captured by the selenium command to process.
    */
-  async processScreenshot(context, base64Screenshot) {
+  async processScreenshot(_context, _base64Screenshot) {
     return Promise.resolve();
   }
 
+  /**
+   * Creates the sanitized test name for the screenshot.
+   * @param {String} fullName - test name
+   * returns {String} test name
+   */
+  // eslint-disable-next-line class-methods-use-this
+  createTestName(fullName) {
+    const matches = TEST_ID_REGEX.exec(fullName);
+
+    // If test ID is provided, use the ID for a shorter test name, otherwise use the full name
+    let name = matches ? matches[1] : fullName.trim();
+
+    // Remove white space
```

Comment:
Is this removing whitespaces and replacing it with an underscore?

Reply:
Yes it would change `Default Screenshot` to `Default_Screenshot`